lib/msf/core/constants.rb

```diff
@@ -49,6 +49,49 @@ module Msf
     ExcellentRanking => "excellent"
   }
 
+#
+# Stability traits
+#
+
+# Module does not crash the service
+CRASH_SAFE             = 'crash-safe'
+# Module crashes service, but service restarts.
+CRASH_SERVICE_RESTARTS = 'crash-service-restarts'
+# Module crashes service, and service remains down.
+CRASH_SERVICE_DEAD     = 'crash-service-dead'
+# Module crashes the OS, but OS restarts.
+CRASH_OS_RESTARTS      = 'crash-os-restarts'
+# Module crashes the OS, and OS remains down.
+CRASH_OS_DEAD          = 'crash-os-dead'
+# Module causes a resource (such as a file or data in database) to be unavailable for the service.
+SERVICE_RESOURCE_LOSS  = 'service-resource-loss'
+# Modules causes a resource (such as a file) to be unavailable for the OS.
+OS_RESOURCE_LOSS       = 'os-resource-loss'
+
+#
+# Side-effect traits
+#
+
+# Modules leaves payload or a dropper on the target machine
+PAYLOAD_ON_DISK        = 'payload-on-disk'
+# Module modifies some config file on the target machine
+CONFIG_CHANGES         = 'config-changes'
+# Module leaves attack traces in a log file (Example: SQL injection data found in HTTP log)
+MALICIOUS_INPUT_IN_LOG = 'malicious-input-in-log'
+# Module may cause account lockouts (likely due to brute-forcing)
+LOCKOUTS               = 'lockouts'
+# Module may show something on the screen (Example: a window pops up)
+SCREEN_EFFECTS         = 'screen-effects'
+
+#
+# Reliability
+#
+
+# The module is may fail at first attempt
+FIRST_ATTEMPT_FAIL = 'first-attempt-fail'
```

Comment:
This makes it sound unconditional?

Reply:
I suppose this could be leveraged into a helpful print for the user. :)

Reply:
I'm open to suggestions, really. I agree some conditions are kind of tough to define. Sometimes maybe some shouldn't even be defined. But that's ok, you can just tell me what you think is right, we can go ahead and do that. If Cody has a different opinion, we can talk about it again. No problem.

Let me know how you'd like to proceed :-)

Reply:
This is still better than `OpsecSafe` in Empire or `Privileged` in Metasploit. :-)

I think this is good progress already. This really is a difficult problem to characterize so much about exploitation in so few words.